### PR TITLE
Improved multiply method

### PR DESCRIPTION
**Pull request by jakwings**

### CODE REVIEW

File: BigInteger.js [hunk 1]
```diff
@@ -146,39 +146,18 @@ var bigInt = (function () {
...
-            if (len > max) max = len;
-        }
-        var result = [], carry = 0;
-        for (var i = 0; i < max || carry > 0; i++) {
-            var sum = carry;
-            for (var j = 0; j < resultSum.length; j++) {
-                sum += resultSum[j][i] || 0;
+                var product = x * y + result[i+j];
+                result[i+j] = product % base;
+                result[i+j+1] += Math.floor(product / base);
```
Comment (Yaffle) [inline BigInteger.js]: Possilby, subtraction works a little faster:
```
                var q = Math.floor(product / base);
                result[i+j] = product - q * base;
                result[i+j+1] += q;
```

Comment (jakwings) [inline BigInteger.js]: Maybe. I hope someone would like to make a benchmark script for these methods… :-P

Comment (Yaffle) [inline BigInteger.js]: see http://yaffle.github.io/BigInteger/

Comment (jakwings) [inline BigInteger.js]: Great. It is indeed faster. :+1: Your ES6 project is great, too.

**Comment by Yaffle** [comment]
@peterolson, 
LGTM

File: BigInteger.js (hunk 1 above)
Comment (Yaffle) [inline BigInteger.js]: Thanks, 
I have updated this lib in my benchmark.

**Comment by peterolson** [comment]
Thanks, it looks good to me too.

**Pull request merged**